# Superstore Data Analysis Project

## Phase 1 — Business Understanding

### 1.1 Business Problem

Superstore generates significant sales, but management is concerned that strong revenue may be hiding profitability problems. The company wants to understand where the business is performing well, where it is losing money, and what actions could improve profitability without damaging sustainable sales growth.

### Project Goal

The goal of this project is to analyze Superstore's sales and profitability performance, identify the key factors driving profits and losses, and provide data-driven recommendations that can help management improve profitability while maintaining sustainable sales growth.

### 1.2 Stakeholders

The key stakeholders for this analysis are:

- **Executive Management:** Requires an overall view of business performance and insights to support strategic decision-making.
- **Sales Management:** Needs to understand sales performance across regions, customer segments, and product categories.
- **Finance Management:** Focuses on profitability, profit margins, losses, and the financial impact of discounts.
- **Regional Managers:** Need to identify high-performing and underperforming regions, states, and markets.
- **Product and Category Managers:** Need insights into which products and categories are driving sales and profitability and which are generating losses.
- **Marketing Management:** Can use customer and segment-level insights to improve targeting and promotional strategies.

### 1.3 Analysis Objectives

The objectives of this analysis are to:

- Evaluate the overall sales and profitability performance of Superstore.
- Analyze sales and profit trends over time to identify growth patterns and periods of underperformance.
- Compare performance across regions, states, and cities to identify strong and weak geographic markets.
- Evaluate the performance of product categories, sub-categories, and individual products.
- Identify products, locations, and customer segments that contribute significantly to profits or losses.
- Investigate the relationship between discounts and profitability.
- Analyze customer segments to understand their contribution to sales and profit.
- Identify key patterns and opportunities that could help improve business performance.
- Develop data-driven recommendations to improve profitability while supporting sustainable sales growth.
- Analyze shipping performance by examining delivery times and shipping modes.
- Analyze customer purchasing behavior and identify high-value customers.
- Distinguish between transaction line-level metrics and order-level metrics to ensure accurate calculations.

### 1.4 Analytical Business Questions

The analysis will seek to answer the following business questions:

#### Overall Business Performance
1. What are the total sales, total profit, total orders, quantity sold, average order value, and overall profit margin?
2. How have sales and profits changed over time?
3. Is Superstore achieving sales growth while maintaining profitability?

#### Geographic Performance
4. Which regions generate the highest sales and profits?
5. Which states and cities are the strongest and weakest performers?
6. Which geographic markets generate high sales but low or negative profits?

#### Product Performance
7. Which product categories and sub-categories generate the most sales and profit?
8. Which products are the most and least profitable?
9. Which products generate high sales but poor profitability?
10. Which products and sub-categories contribute most significantly to overall losses?

#### Customer and Segment Performance
11. Which customer segments generate the most sales and profit?
12. Who are Superstore's highest-value customers?
13. Which customers generate high sales but relatively low or negative profits?
14. How frequently do customers purchase from Superstore, and are there identifiable groups of high-value or at-risk customers?

#### Discount and Profitability Analysis
15. What is the relationship between discounts and profitability?
16. At what discount levels do transactions become more likely to generate losses?
17. Which products, categories, customer segments, or regions are most negatively affected by discounting?

#### Shipping Performance
18. How long does Superstore take to ship orders?
19. How does shipping time vary by shipping mode?
20. Are there noticeable relationships between shipping mode, sales, and profitability?

#### Trends and Opportunities
21. Are there seasonal patterns in sales and profitability?
22. Are high sales consistently associated with high profits?
23. What characteristics are common among loss-making transactions?
24. Where are the greatest opportunities for Superstore to improve profitability while maintaining sustainable sales growth?

### 1.5 Key Performance Indicators (KPIs)

The following KPIs will be used to evaluate Superstore's business performance:

| KPI | Formula | Purpose |
|---|---|---|
| Total Sales | Sum of Sales | Measures total revenue generated |
| Total Profit | Sum of Profit | Measures overall profitability |
| Total Orders | Count of unique Order IDs | Measures the number of distinct customer orders |
| Total Quantity Sold | Sum of Quantity | Measures the total number of units sold |
| Average Order Value (AOV) | Total Sales / Total Orders | Measures the average revenue generated per order |
| Overall Profit Margin | Total Profit / Total Sales × 100 | Measures how much profit is generated from each unit of sales |
| Average Discount | Sum of Discount / Number of transaction lines | Measures the average discount applied across transaction lines |
| Sales Growth Rate | (Current Period Sales - Previous Period Sales) / Previous Period Sales × 100 | Measures the percentage change in sales between periods |
| Profit Growth Rate | (Current Period Profit - Previous Period Profit) / Previous Period Profit × 100 | Measures the percentage change in profit between periods |
| Average Shipping Time | Sum of Shipping Days / Number of Orders | Measures the average time between order and shipment dates |
| Loss-Making Transaction Rate | Number of loss-making transaction lines / Total transaction lines × 100 | Measures the proportion of transaction lines generating losses |

### 1.6 Decisions the Analysis Should Support

The analysis should help Superstore management make better decisions about:

- **Products:** Identify which products and categories to promote, review, or improve.
- **Discounts:** Determine where discounts may be contributing to losses and where discount policies should be reviewed.
- **Geographic Markets:** Identify high-performing and underperforming regions, states, and cities.
- **Customers:** Identify valuable customers and customer segments that contribute strongly to the business.
- **Sales Strategy:** Identify opportunities to grow sales without sacrificing profitability.
- **Shipping:** Understand shipping performance and identify possible areas for improvement.
- **Resource Allocation:** Help management focus attention and resources on the areas with the greatest potential for profitable growth.

### 1.7: Initial Dataset Inspection

In [1]:
import pandas as pd

In [5]:
df = pd.read_csv("../data/raw/Sample - Superstore.csv", encoding="latin1")

In [6]:
df.head()

,Row ID,Order ID,Order Date,Ship Date,Ship Mode,Customer ID,Customer Name,Segment,Country,City,...,Postal Code,Region,Product ID,Category,Sub-Category,Product Name,Sales,Quantity,Discount,Profit
0,1,CA-2016-152156,11/8/2016,11/11/2016,Second Class,CG-12520,Claire Gute,Consumer,United States,Henderson,...,42420,South,FUR-BO-10001798,Furniture,Bookcases,Bush Somerset Collection Bookcase,261.9600,2,0.00,41.9136
1,2,CA-2016-152156,11/8/2016,11/11/2016,Second Class,CG-12520,Claire Gute,Consumer,United States,Henderson,...,42420,South,FUR-CH-10000454,Furniture,Chairs,"Hon Deluxe Fabric Upholstered Stacking Chairs,...",731.9400,3,0.00,219.5820
2,3,CA-2016-138688,6/12/2016,6/16/2016,Second Class,DV-13045,Darrin Van Huff,Corporate,United States,Los Angeles,...,90036,West,OFF-LA-10000240,Office Supplies,Labels,Self-Adhesive Address Labels for Typewriters b...,14.6200,2,0.00,6.8714
3,4,US-2015-108966,10/11/2015,10/18/2015,Standard Class,SO-20335,Sean O'Donnell,Consumer,United States,Fort Lauderdale,...,33311,South,FUR-TA-10000577,Furniture,Tables,Bretford CR4500 Series Slim Rectangular Table,957.5775,5,0.45,-383.0310
4,5,US-2015-108966,10/11/2015,10/18/2015,Standard Class,SO-20335,Sean O'Donnell,Consumer,United States,Fort Lauderdale,...,33311,South,OFF-ST-10000760,Office Supplies,Storage,Eldon Fold 'N Roll Cart System,22.3680,2,0.20,2.5164


In [8]:
df.shape

(9994, 21)

In [9]:
df.columns

Index(['Row ID', 'Order ID', 'Order Date', 'Ship Date', 'Ship Mode',
       'Customer ID', 'Customer Name', 'Segment', 'Country', 'City', 'State',
       'Postal Code', 'Region', 'Product ID', 'Category', 'Sub-Category',
       'Product Name', 'Sales', 'Quantity', 'Discount', 'Profit'],
      dtype='str')

In [10]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 9994 entries, 0 to 9993
Data columns (total 21 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Row ID         9994 non-null   int64  
 1   Order ID       9994 non-null   str    
 2   Order Date     9994 non-null   str    
 3   Ship Date      9994 non-null   str    
 4   Ship Mode      9994 non-null   str    
 5   Customer ID    9994 non-null   str    
 6   Customer Name  9994 non-null   str    
 7   Segment        9994 non-null   str    
 8   Country        9994 non-null   str    
 9   City           9994 non-null   str    
 10  State          9994 non-null   str    
 11  Postal Code    9994 non-null   int64  
 12  Region         9994 non-null   str    
 13  Product ID     9994 non-null   str    
 14  Category       9994 non-null   str    
 15  Sub-Category   9994 non-null   str    
 16  Product Name   9994 non-null   str    
 17  Sales          9994 non-null   float64
 18  Quantity       9994

### 1.8 Data Dictionary

| Column | Description | Type |
|---|---|---|
| Row ID | Unique identifier for each dataset row | ID |
| Order ID | Unique identifier for each customer order | ID |
| Order Date | Date the customer placed the order | Date |
| Ship Date | Date the order was shipped | Date |
| Ship Mode | Shipping method used | Categorical |
| Customer ID | Unique identifier for each customer | ID |
| Customer Name | Name of the customer | Categorical |
| Segment | Customer segment | Categorical |
| Country | Country where the order was placed | Categorical |
| City | Customer/order city | Categorical |
| State | Customer/order state | Categorical |
| Postal Code | Postal code of the location | ID / Geographic |
| Region | Geographic region | Categorical |
| Product ID | Unique identifier for a product | ID |
| Category | Main product category | Categorical |
| Sub-Category | Product sub-category | Categorical |
| Product Name | Name of the product | Categorical |
| Sales | Sales revenue generated by the transaction line | Numerical |
| Quantity | Number of units sold | Numerical |
| Discount | Discount applied to the transaction line | Numerical |
| Profit | Profit generated by the transaction line | Numerical |

### 1.9 Initial Observations and Assumptions

#### Initial Observations

- The dataset contains 9,994 transaction-line records and 21 columns.
- The dataset is structured at the order-line level, meaning one order may appear across multiple rows when it contains multiple products.
- The data contains information about orders, customers, products, geographic locations, shipping, sales, discounts, quantities, and profit.
- The dataset covers multiple geographic regions and product categories.
- Sales and Profit are provided, but there is no separate cost-of-goods column.
- Order Date and Ship Date can be used to analyze sales trends and shipping times.
- The raw CSV requires explicit character encoding when loaded into Pandas.

#### Initial Assumptions

- Each Order ID represents a unique customer order.
- Each Customer ID represents a unique customer.
- Each Product ID represents a product identifier.
- Sales represents revenue generated by a transaction line.
- Profit represents the profit associated with a transaction line.
- Discount is represented as a decimal value, where 0.20 represents a 20% discount.
- Negative profit values represent loss-making transaction lines.
- The analysis will treat the provided data as historical business transaction data.
- These assumptions will be validated where possible during the data quality assessment.

In [1]:
import pandas as pd


In [2]:
df = pd.read_csv("../data/raw/Sample - Superstore.csv", encoding="latin1")

## Phase 2 — Data Understanding & Quality Assessment

### 2.1 Dataset Structure and Data Types

In [3]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 9994 entries, 0 to 9993
Data columns (total 21 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Row ID         9994 non-null   int64  
 1   Order ID       9994 non-null   str    
 2   Order Date     9994 non-null   str    
 3   Ship Date      9994 non-null   str    
 4   Ship Mode      9994 non-null   str    
 5   Customer ID    9994 non-null   str    
 6   Customer Name  9994 non-null   str    
 7   Segment        9994 non-null   str    
 8   Country        9994 non-null   str    
 9   City           9994 non-null   str    
 10  State          9994 non-null   str    
 11  Postal Code    9994 non-null   int64  
 12  Region         9994 non-null   str    
 13  Product ID     9994 non-null   str    
 14  Category       9994 non-null   str    
 15  Sub-Category   9994 non-null   str    
 16  Product Name   9994 non-null   str    
 17  Sales          9994 non-null   float64
 18  Quantity       9994

In [4]:
df.dtypes

Row ID             int64
Order ID             str
Order Date           str
Ship Date            str
Ship Mode            str
Customer ID          str
Customer Name        str
Segment              str
Country              str
City                 str
State                str
Postal Code        int64
Region               str
Product ID           str
Category             str
Sub-Category         str
Product Name         str
Sales            float64
Quantity           int64
Discount         float64
Profit           float64
dtype: object

### 2.2 Data Type Assessment

- The dataset contains numerical, categorical, identifier, geographic, and date variables.
- Sales, Quantity, Discount, and Profit are stored as numerical data types.
- Order Date and Ship Date are currently stored as object (text) rather than datetime values.
- The date columns will need to be converted to datetime during the data cleaning phase.
- Identifier columns such as Order ID, Customer ID, and Product ID should be treated as identifiers rather than numerical measures.
- Postal Code should be treated as a geographic identifier rather than a value used for mathematical calculations.

In [5]:
df.isnull().sum()

Row ID           0
Order ID         0
Order Date       0
Ship Date        0
Ship Mode        0
Customer ID      0
Customer Name    0
Segment          0
Country          0
City             0
State            0
Postal Code      0
Region           0
Product ID       0
Category         0
Sub-Category     0
Product Name     0
Sales            0
Quantity         0
Discount         0
Profit           0
dtype: int64

### 2.3 Missing Values Assessment

The dataset was checked for missing values across all 21 columns.

- No missing values were identified in the dataset.
- Therefore, no missing-value treatment is currently required.
- The dataset appears complete with respect to null values.

In [6]:
df.duplicated().sum()

np.int64(0)

In [7]:
df["Row ID"].duplicated().sum()

np.int64(0)

In [8]:
df["Order ID"].duplicated().sum()

np.int64(4985)

In [9]:
df[df["Order ID"].duplicated(keep=False)].sort_values("Order ID").head(10)

,Row ID,Order ID,Order Date,Ship Date,Ship Mode,Customer ID,Customer Name,Segment,Country,City,...,Postal Code,Region,Product ID,Category,Sub-Category,Product Name,Sales,Quantity,Discount,Profit
6288,6289,CA-2014-100090,7/8/2014,7/12/2014,Standard Class,EB-13705,Ed Braxton,Corporate,United States,San Francisco,...,94122,West,OFF-BI-10001597,Office Supplies,Binders,"Wilson Jones Ledger-Size, Piano-Hinge Binder, ...",196.704,6,0.2,68.8464
6287,6288,CA-2014-100090,7/8/2014,7/12/2014,Standard Class,EB-13705,Ed Braxton,Corporate,United States,San Francisco,...,94122,West,FUR-TA-10003715,Furniture,Tables,Hon 2111 Invitation Series Corner Table,502.488,3,0.2,-87.9354
3835,3836,CA-2014-100363,4/8/2014,4/15/2014,Standard Class,JM-15655,Jim Mitchum,Corporate,United States,Glendale,...,85301,West,OFF-FA-10000611,Office Supplies,Fasteners,Binder Clips by OIC,2.368,2,0.2,0.8288
3836,3837,CA-2014-100363,4/8/2014,4/15/2014,Standard Class,JM-15655,Jim Mitchum,Corporate,United States,Glendale,...,85301,West,OFF-PA-10004733,Office Supplies,Paper,Things To Do Today Spiral Book,19.008,3,0.2,6.8904
6571,6572,CA-2014-100678,4/18/2014,4/22/2014,Standard Class,KM-16720,Kunst Miller,Consumer,United States,Houston,...,77095,Central,TEC-AC-10000474,Technology,Accessories,Kensington Expert Mouse Optical USB Trackball ...,227.976,3,0.2,28.4970
6568,6569,CA-2014-100678,4/18/2014,4/22/2014,Standard Class,KM-16720,Kunst Miller,Consumer,United States,Houston,...,77095,Central,OFF-AR-10001868,Office Supplies,Art,Prang Dustless Chalk Sticks,2.688,2,0.2,1.0080
6569,6570,CA-2014-100678,4/18/2014,4/22/2014,Standard Class,KM-16720,Kunst Miller,Consumer,United States,Houston,...,77095,Central,FUR-CH-10002602,Furniture,Chairs,DMI Arturo Collection Mission-style Design Woo...,317.058,3,0.3,-18.1176
6570,6571,CA-2014-100678,4/18/2014,4/22/2014,Standard Class,KM-16720,Kunst Miller,Consumer,United States,Houston,...,77095,Central,OFF-EN-10000056,Office Supplies,Envelopes,Cameo Buff Policy Envelopes,149.352,3,0.2,50.4063
9019,9020,CA-2014-100706,12/16/2014,12/18/2014,Second Class,LE-16810,Laurel Elliston,Consumer,United States,Springfield,...,22153,South,TEC-AC-10001314,Technology,Accessories,Case Logic 2.4GHz Wireless Keyboard,99.980,2,0.0,7.9984
9020,9021,CA-2014-100706,12/16/2014,12/18/2014,Second Class,LE-16810,Laurel Elliston,Consumer,United States,Springfield,...,22153,South,FUR-FU-10002268,Furniture,Furnishings,Ultra Door Push Plate,29.460,6,0.0,9.7218


### 2.4 Duplicate Data Assessment

The dataset was examined for exact duplicate records and repeated identifier values to determine whether duplicate data could affect the analysis.

#### Exact Duplicate Rows

- No exact duplicate rows were identified in the dataset.
- This indicates that no two records contain identical values across all 21 columns.
- Therefore, no rows need to be removed on the basis of being complete duplicates.

#### Row ID Validation

- No duplicate values were identified in the **Row ID** column.
- Each of the 9,994 transaction-line records has a unique Row ID.
- Row ID can therefore be used to uniquely identify individual records within the dataset.

#### Order ID Assessment

- The duplicate check identified **4,985 repeated occurrences** in the Order ID column.
- Inspection of the repeated Order IDs shows that the same Order ID can appear across multiple rows with different products, categories, sales values, quantities, and profit values.
- This is expected because a single customer order may contain multiple products, with each product recorded as a separate transaction line.
- Repeated Order IDs are therefore valid records and should not be removed as duplicates.

#### Conclusion

The dataset contains no exact duplicate records and no duplicated Row IDs. Repeated Order IDs reflect the order-line structure of the dataset rather than a data-quality problem. Therefore, no duplicate records require removal based on the current assessment.

In [10]:
categorical_columns = [
    "Ship Mode",
    "Segment",
    "Country",
    "Region",
    "Category",
    "Sub-Category"
]

for column in categorical_columns:
    print(f"\n{column}:")
    print(df[column].unique())


Ship Mode:
<StringArray>
['Second Class', 'Standard Class', 'First Class', 'Same Day']
Length: 4, dtype: str

Segment:
<StringArray>
['Consumer', 'Corporate', 'Home Office']
Length: 3, dtype: str

Country:
<StringArray>
['United States']
Length: 1, dtype: str

Region:
<StringArray>
['South', 'West', 'Central', 'East']
Length: 4, dtype: str

Category:
<StringArray>
['Furniture', 'Office Supplies', 'Technology']
Length: 3, dtype: str

Sub-Category:
<StringArray>
[  'Bookcases',      'Chairs',      'Labels',      'Tables',     'Storage',
 'Furnishings',         'Art',      'Phones',     'Binders',  'Appliances',
       'Paper', 'Accessories',   'Envelopes',   'Fasteners',    'Supplies',
    'Machines',     'Copiers']
Length: 17, dtype: str


In [11]:
df[categorical_columns].nunique()

Ship Mode        4
Segment          3
Country          1
Region           4
Category         3
Sub-Category    17
dtype: int64

In [12]:
df["Region"].value_counts()

Region
West       3203
East       2848
Central    2323
South      1620
Name: count, dtype: int64

In [13]:
df["Category"].value_counts()

Category
Office Supplies    6026
Furniture          2121
Technology         1847
Name: count, dtype: int64

In [15]:
df["Ship Mode"].value_counts()

Ship Mode
Standard Class    5968
Second Class      1945
First Class       1538
Same Day           543
Name: count, dtype: int64

### 2.5 Categorical Data Assessment

The main categorical variables were examined using unique-value and frequency analysis to identify inconsistent spelling, capitalization, unexpected categories, and the distribution of records across major groups.

#### Category Structure

- **Ship Mode** contains **4 unique values**: Standard Class, Second Class, First Class, and Same Day.
- **Segment** contains **3 unique values**: Consumer, Corporate, and Home Office.
- **Country** contains **1 unique value**: United States.
- **Region** contains **4 unique values**: West, East, Central, and South.
- **Category** contains **3 unique values**: Office Supplies, Furniture, and Technology.
- **Sub-Category** contains **17 unique values**, including Binders, Paper, Furnishings, Phones, Storage, Chairs, Copiers, Machines, and others.

#### Frequency Observations

- The **West** region has the highest number of transaction-line records with **3,203**, followed by East with **2,848**, Central with **2,323**, and South with **1,620**.
- **Office Supplies** is the most frequently occurring product category with **6,026 transaction lines**, followed by Furniture with **2,121** and Technology with **1,847**.
- **Standard Class** is the most commonly used shipping mode with **5,968 transaction lines**, followed by Second Class with **1,945**, First Class with **1,538**, and Same Day with **543**.
- The dataset covers only the **United States**, so geographic analysis will be limited to locations within one country.

#### Data Quality Assessment

- No obvious inconsistencies in spelling or capitalization were identified among the categorical values examined.
- The categorical values appear to be standardized and suitable for grouping and comparison during analysis.
- No category values require correction based on the initial assessment.
- The differences in frequency represent the number of **transaction lines** in each group and should not automatically be interpreted as differences in sales, profit, or the number of unique orders.

In [16]:
df["Discount"].min(), df["Discount"].max()

(np.float64(0.0), np.float64(0.8))

In [17]:
df[(df["Discount"] < 0) | (df["Discount"] > 1)]

,Row ID,Order ID,Order Date,Ship Date,Ship Mode,Customer ID,Customer Name,Segment,Country,City,...,Postal Code,Region,Product ID,Category,Sub-Category,Product Name,Sales,Quantity,Discount,Profit


In [18]:
df[df["Sales"] <= 0]

,Row ID,Order ID,Order Date,Ship Date,Ship Mode,Customer ID,Customer Name,Segment,Country,City,...,Postal Code,Region,Product ID,Category,Sub-Category,Product Name,Sales,Quantity,Discount,Profit


In [19]:
df[df["Quantity"] <= 0]

,Row ID,Order ID,Order Date,Ship Date,Ship Mode,Customer ID,Customer Name,Segment,Country,City,...,Postal Code,Region,Product ID,Category,Sub-Category,Product Name,Sales,Quantity,Discount,Profit


In [20]:
(df["Profit"] < 0).sum()

np.int64(1871)

In [21]:
order_date_check = pd.to_datetime(df["Order Date"])
ship_date_check = pd.to_datetime(df["Ship Date"])

In [22]:
(ship_date_check < order_date_check).sum()

np.int64(0)

### 2.6 Business-Rule Validation

The dataset was tested against basic business rules to identify invalid or suspicious records.

- Discount values range from 0% to 80%, with no values outside the expected 0 to 1 range.
- No transaction lines have zero or negative sales.
- No transaction lines have zero or negative quantities.
- 1,871 transaction lines have negative profit values. These represent loss-making transactions rather than data-quality errors and will be investigated further during the analysis.
- No records have a Ship Date earlier than the corresponding Order Date.

Overall, no obvious business-rule violations were identified in the tested fields.

In [24]:
next
df[['Sales', 'Quantity', 'Discount', 'Profit']].describe() 

,Sales,Quantity,Discount,Profit
count,9994.000000,9994.000000,9994.000000,9994.000000
mean,229.858001,3.789574,0.156203,28.656896
std,623.245101,2.225110,0.206452,234.260108
min,0.444000,1.000000,0.000000,-6599.978000
25%,17.280000,2.000000,0.000000,1.728750
50%,54.490000,3.000000,0.200000,8.666500
75%,209.940000,5.000000,0.200000,29.364000
max,22638.480000,14.000000,0.800000,8399.976000


In [25]:
df.nlargest(10, "Sales")[
    ["Order ID", "Product Name", "Category", "Sales", "Discount", "Profit"]
]

,Order ID,Product Name,Category,Sales,Discount,Profit
2697,CA-2014-145317,Cisco TelePresence System EX90 Videoconferenci...,Technology,22638.480,0.5,-1811.0784
6826,CA-2016-118689,Canon imageCLASS 2200 Advanced Copier,Technology,17499.950,0.0,8399.9760
8153,CA-2017-140151,Canon imageCLASS 2200 Advanced Copier,Technology,13999.960,0.0,6719.9808
2623,CA-2017-127180,Canon imageCLASS 2200 Advanced Copier,Technology,11199.968,0.2,3919.9888
4190,CA-2017-166709,Canon imageCLASS 2200 Advanced Copier,Technology,10499.970,0.0,5039.9856
9039,CA-2016-117121,GBC Ibimaster 500 Manual ProClick Binding System,Office Supplies,9892.740,0.0,4946.3700
4098,CA-2014-116904,Ibico EPK-21 Electric Binding System,Office Supplies,9449.950,0.0,4630.4755
4277,US-2016-107440,"3D Systems Cube Printer, 2nd Generation, Magenta",Technology,9099.930,0.0,2365.9818
8488,CA-2016-158841,HP Designjet T520 Inkjet Large Format Printer ...,Technology,8749.950,0.0,2799.9840
6425,CA-2016-143714,Canon imageCLASS 2200 Advanced Copier,Technology,8399.976,0.4,1119.9968


In [30]:
df.nsmallest(10, "Sales")[
    ["Order ID", "Product Name", "Category", "Sales", "Discount", "Profit"]
]

,Order ID,Product Name,Category,Sales,Discount,Profit
4101,US-2017-102288,Hoover Replacement Belt for Commercial Guardsm...,Office Supplies,0.444,0.8,-1.1100
9292,CA-2017-124114,Acco Suede Grain Vinyl Round Ring Binder,Office Supplies,0.556,0.8,-0.9452
8658,CA-2016-168361,Avery Durable Slant Ring Binders With Label Ho...,Office Supplies,0.836,0.8,-1.3376
4711,CA-2014-112403,Avery Round Ring Poly Binders,Office Supplies,0.852,0.7,-0.5964
2106,US-2014-152723,Acco 3-Hole Punch,Office Supplies,0.876,0.8,-1.4016
7548,CA-2014-103492,Avery Non-Stick Binders,Office Supplies,0.898,0.8,-1.5715
8033,CA-2015-119690,"Avery Triangle Shaped Sheet Lifters, Black, 2/...",Office Supplies,0.984,0.8,-1.4760
2761,CA-2017-126536,Maxell 4.7GB DVD-R 5/Pack,Technology,0.990,0.0,0.4356
8024,CA-2014-129189,Acco Economy Flexible Poly Round Ring Binder,Office Supplies,1.044,0.8,-1.8270
976,US-2017-100209,Wilson Jones Easy Flow II Sheet Lifters,Office Supplies,1.080,0.7,-0.7920


In [31]:

df.nlargest(10, "Profit")[
    ["Order ID", "Product Name", "Category", "Sales", "Discount", "Profit"]
]

,Order ID,Product Name,Category,Sales,Discount,Profit
6826,CA-2016-118689,Canon imageCLASS 2200 Advanced Copier,Technology,17499.950,0.0,8399.9760
8153,CA-2017-140151,Canon imageCLASS 2200 Advanced Copier,Technology,13999.960,0.0,6719.9808
4190,CA-2017-166709,Canon imageCLASS 2200 Advanced Copier,Technology,10499.970,0.0,5039.9856
9039,CA-2016-117121,GBC Ibimaster 500 Manual ProClick Binding System,Office Supplies,9892.740,0.0,4946.3700
4098,CA-2014-116904,Ibico EPK-21 Electric Binding System,Office Supplies,9449.950,0.0,4630.4755
2623,CA-2017-127180,Canon imageCLASS 2200 Advanced Copier,Technology,11199.968,0.2,3919.9888
509,CA-2015-145352,Fellowes PB500 Electric Punch Plastic Comb Bin...,Office Supplies,6354.950,0.0,3177.4750
8488,CA-2016-158841,HP Designjet T520 Inkjet Large Format Printer ...,Technology,8749.950,0.0,2799.9840
7666,US-2016-140158,Hewlett Packard LaserJet 3310 Copier,Technology,5399.910,0.0,2591.9568
6520,CA-2017-138289,GBC DocuBind P400 Electric Binding System,Office Supplies,5443.960,0.0,2504.2216


In [32]:
df.nsmallest(10, "Profit")[
    ["Order ID", "Product Name", "Category", "Sales", "Discount", "Profit"]
]

,Order ID,Product Name,Category,Sales,Discount,Profit
7772,CA-2016-108196,Cubify CubeX 3D Printer Double Head Print,Technology,4499.985,0.7,-6599.9780
683,US-2017-168116,Cubify CubeX 3D Printer Triple Head Print,Technology,7999.980,0.5,-3839.9904
9774,CA-2014-169019,GBC DocuBind P400 Electric Binding System,Office Supplies,2177.584,0.8,-3701.8928
3011,CA-2017-134845,Lexmark MX611dhe Monochrome Laser Printer,Technology,2549.985,0.7,-3399.9800
4991,US-2017-122714,Ibico EPK-21 Electric Binding System,Office Supplies,1889.990,0.8,-2929.4845
3151,CA-2015-147830,Cubify CubeX 3D Printer Double Head Print,Technology,1799.994,0.7,-2639.9912
5310,CA-2017-131254,Fellowes PB500 Electric Punch Plastic Comb Bin...,Office Supplies,1525.188,0.8,-2287.7820
9639,CA-2015-116638,Chromcraft Bull-Nose Wood Oval Conference Tabl...,Furniture,4297.644,0.4,-1862.3124
1199,CA-2016-130946,GBC DocuBind P400 Electric Binding System,Office Supplies,1088.792,0.8,-1850.9464
2697,CA-2014-145317,Cisco TelePresence System EX90 Videoconferenci...,Technology,22638.480,0.5,-1811.0784


### 2.7 Numerical Values and Potential Outliers

The numerical variables were examined using summary statistics and extreme-value analysis to identify unusually high or low values and determine whether they represent potential data-quality issues.

#### Sales Observations

- The maximum transaction-line Sales value is **22,638.48**, while the minimum is **0.44**.
- The median Sales value is **54.49**, compared with a 75th percentile value of **209.94** and a maximum of **22,638.48**. This indicates that a relatively small number of transactions have substantially higher sales than a typical transaction.
- The highest-sales transaction was for a **Cisco TelePresence System** in the **Technology** category, generating **22,638.48 in Sales**.
- Most of the highest-sales transactions involve high-value products from the **Technology** category, particularly copiers and printing equipment. Some high-value **Office Supplies** products also appear among the largest transactions.
- The smallest sales transactions are mainly low-value **Office Supplies** products.

#### Profit and Loss Observations

- The highest transaction-line Profit recorded is **8,399.98**, generated by a **Canon imageCLASS 2200 Advanced Copier** in the **Technology** category with **17,499.95 in Sales** and no discount.
- The largest transaction-line loss is **-6,599.98**, generated by a **Cubify CubeX 3D Printer** in the **Technology** category. The transaction had **4,499.99 in Sales** and received a **70% discount**.
- Several of the highest-profit transactions are associated with **Technology products**, particularly copiers.
- The largest losses are concentrated among high-value products from the **Technology** and **Office Supplies** categories, with one **Furniture** transaction also appearing among the ten largest losses.

#### Discount Observations

- A noticeable pattern appears among the largest loss-making transactions: many received substantial discounts ranging from **40% to 80%**.
- The largest loss received a **70% discount**, while several other major losses received discounts of **50%, 70%, or 80%**.
- In contrast, many of the highest-profit transactions received **no discount**, although at least one highly profitable transaction remained profitable with a **20% discount**.
- The highest-sales transaction received a **50% discount** and generated a loss of **1,811.08**, showing that high sales do not necessarily result in high profit.
- These observations suggest a possible relationship between high discount levels and reduced profitability. However, the extreme transactions alone are not sufficient to prove that discounts cause losses. The relationship will be tested across the full dataset during further analysis.

#### Outlier Assessment

- The dataset contains extreme Sales and Profit values compared with typical transaction values.
- However, the extreme values appear to be associated with identifiable high-value products rather than obvious data-entry errors.
- These potential outliers will therefore be retained rather than automatically removed.
- Further analysis will determine how much these extreme transactions influence overall sales and profitability results.

### 2.8 Data Suitability and Limitations

The dataset is suitable for answering the main business questions related to sales, profitability, customers, products, geographic performance, discounts, and shipping patterns.

The available data allows the analysis to examine:

- Sales and profit performance over time using Order Date.
- Performance across regions, states, and cities.
- Product performance across categories, sub-categories, and individual products.
- Customer and customer-segment performance.
- The relationship between discounts and profitability.
- Shipping preparation time using Order Date and Ship Date.
- High-performing and loss-making transactions.

However, the dataset has several limitations:

- No detailed cost information is available beyond the provided Profit values.
- No marketing or customer acquisition cost data is available.
- No inventory or stock-level information is available.
- No actual delivery date is provided, so shipping analysis is limited to the number of days between Order Date and Ship Date.
- No explicit returns information is available.
- The dataset contains historical transaction data, so findings describe historical performance and do not automatically predict future performance.

These limitations will be considered when interpreting findings and making recommendations.